File: empl/Python/Employe.py
class Employe:
    def __init__(self, nom, age, salaire, numero_securite_sociale):
        self.nom = nom
        self.age = age
        self.salaire = salaire
        self.numero_securite_sociale = numero_securite_sociale

    def afficher(self):
        return f"Nom: {self.nom}, Age: {self.age}, Salaire: {self.salaire}, N° Sécurité Sociale: {self.numero_securite_sociale}"

    def augmentation(self, pourcentage):
        return self.salaire * (pourcentage / 100)

    def calculer_salaire(self):
        return self.salaire * 0.765  # après déduction des charges sociales

    def __repr__(self):
        return f"Nom: {self.nom}, Age: {self.age}, Salaire: {self.salaire}, N° Sécurité Sociale: {self.numero_securite_sociale}"

File: empl/php/Employe.php
<?php
class Employe
{
    protected string $nom;
    protected int $age;
    protected float $salaire;
    protected string $NumSecu;

    public function __construct($nom, $age, $salaire, $NumSecu)
    {
        $this->nom = $nom;
        $this->age = $age;
        $this->salaire = $salaire;
        $this->NumSecu = $NumSecu;
    }

    public function __get($name): mixed
    {
        return $this->$name;
    }

    public function __set($name, $value): void
    {
        $this->$name = $value;
    }

    public function calculer_salaire(): float
    {
        return $this->salaire * 0.756;
    }

    public function augmentation(float $pourcentage): void
    {
        $this->salaire += $this->salaire * ($pourcentage / 100);
    }
    public function afficher(): string
    {
        return "Nom: {$this->nom}, Age: {$this->age}, Salaire Brut: {$this->salaire}, NumSecu: {$this->NumSecu}";
    }

    public function __toString(): string
    {
        return $this->afficher();
    }
}
